on-premise-docker-registry/README.md

```diff
@@ -0,0 +1,191 @@
+
+- Generate the self-signed certificate for enabling HTTPS for the Docker registry. Be sure to specify the `CN` as `myregistry.kyma`
+
+```shell
+make generate-self-signed-cert
+```
+
+- Add the newly created certificate to your trust storage
+
+```shell
+make trust-self-signed-cert
+```
+
+- Generate the `htpasswd`. This is used for accessing the on-premise Docker registry.
+
+```shell
+make generate-htpasswd
+```
+
+- Start the Docker registry server
+
+```shell
+make start-docker-registry
+```
+
+- Add DNS entry to the `/etc/hosts` file
+
+```shell
+127.0.0.1 myregistry.kyma
+```
+
+- Configure Cloud Connector to access the on-premise Docker registry.
+  ![cc-config](assets/cc-config.png)
+
+## nginx as reverse proxy
+
+We use Nginx as a reverse proxy to forward the HTTP requests for pulling Docker images from the on-premise Docker registry using Connectivity Proxy and Cloud Connector.
+
+It is exposed as the `NodePort` service. This exposes the Service on the Kubernetes worker node on a port. Kubelet is a component running on each Kubernetes worker node. Among other tasks, it is responsible for pulling the Docker images.
+
+When creating a deployment, we specify the Docker registry as `localhost:{NodePort}`. This is the address of the nginx reverse proxy. The nginx reverse proxy then forwards the call to the on-premise Docker registry using Connectivity Proxy and Cloud Connector.
+
+This sample shows a simple configuration with nginx as a reverse proxy. You can use any other reverse proxy implementation based on your on-premise Docker registry behavior and APIs.
+
+- Create namespace and enable istio sidecar injection if not done earlier.
+
+```shell
+kubectl create namespace ${NAMESPACE}
+kubectl label namespace ${NAMESPACE} istio-injection=enabled
+```
+
+- Deploy the nginx as a reverse proxy. The following components will be deployed:
+  - [ConfigMap](./k8s/configmap.yaml) for nginx configuration
+  - [Deployment and Service](./k8s/deployment.yaml)
+  - [PeerAuthentication](./k8s/peer-authentication.yaml) set to `PERMISSIVE` to allow for communication between kubelet and the nginx reverse proxy.
+
+```shell
+make deploy-nginx-reverse-proxy
+```
+
+- Wait for the nginx reverse proxy to be up and running
+
+```shell
+make check-nginx-reverse-proxy
+```
+
+- Export NodePort for the nginx reverse proxy as an environment variable
+
+```shell
+export NGINX_NODE_PORT=$(kubectl get svc nginx -o jsonpath='{.spec.ports[0].nodePort}')
+```
+
+## Test workload
+
+Let's deploy a sample workload that will use an image from the on-premise Docker registry.
+
+- Log in to the created on-premise docker registry
+
+```shell
+make docker-login
+```
+
+- Create and store a sample Docker image in the on-premise Docker registry. It uses an image tag based on the current time.
+
+```shell
```

Comment:
add 3 blank spaces at the beginning of each line of the code block for proper indentation below the list item.